code/__DEFINES/is_helpers.dm

```diff
@@ -20,6 +20,9 @@ GLOBAL_VAR_INIT(magic_appearance_detecting_image, new /image) // appearances are
 GLOBAL_VAR_INIT(refid_filter, TYPEID(filter(type="angular_blur")))
 #define isfilter(thing) (!hascall(thing, "Cut") && TYPEID(thing) == GLOB.refid_filter)
 
+GLOBAL_DATUM_INIT(regex_rgb_text, /regex, regex(@"^#?(([0-9a-fA-F]{8})|([0-9a-fA-F]{6})|([0-9a-fA-F]{3}))"))
+#define iscolortext(thing) (istext(thing) && GLOB.regex_rgb_text.Find(thing) && (length(GLOB.regex_rgb_text.match) == length(thing)))
```

Comment:
Instead of checking the length of the match, use the $ token at the end of the regex to match the end of the string

Reply:
oh yeah, it was a simpler solution